src/objects/api/fields.py

```diff
@@ -42,7 +44,23 @@ def to_internal_value(self, data):
         try:
             return self.get_queryset().get_by_url(data)
         except ObjectDoesNotExist:
-            self.fail("does_not_exist", value=smart_text(data))
+            # if service is configured, but objec_type is missing
+            # let's try to create an ObjectType
+            service = Service.get_service(data)
+            if not service:
+                self.fail("does_not_exist", value=smart_text(data))
```

Comment:
```suggestion
                self.fail("does_not_exist", value=smart_str(data))
```

I think when rebasing on the 4.2 upgrade the import will be updated already